controllers/config.go

```diff
@@ -102,6 +112,54 @@ func InitFromFlags(fs *flag.FlagSet, cfg *ManagerConfig) {
 
 	cfg.SecretControllerConfig = &secret.Config{}
 	secret.InitFromFlags(fs, cfg.SecretControllerConfig)
+
+	return nil
+}
+
+// InitFeatureGates initializes feature gates from the provided CLI flag set.
+func InitFeatureGates(fs *flag.FlagSet, featureFlags *featuregate.MutableFeatureGate) error {
+	*featureFlags = featuregate.NewFeatureGate()
+	if err := (*featureFlags).Add(features.GetDefaultFeatures()); err != nil {
+		return fmt.Errorf("error adding features to the featuregate: %v", err)
+	}
+	(*featureFlags).AddFlag(fs)
+
+	return nil
+}
+
+// PopulateControllerFeatureGates adds relevant feature gates to every controller configuration
+func PopulateControllerFeatureGates(config *ManagerConfig) {
```

Comment:
```suggestion
func populateControllersFeatureGates(config *ManagerConfig) {
```

Same as above, with additional plural (`Controllers`) since this function adds feature gates for all controllers.